specification/dartLangSpec.tex

```diff
@@ -17781,8 +17785,8 @@ \subsection{Switch}
 }
 
 \LMHash{}%
-It is a compile-time error if the operator \lit{==} of class $C$
-is not primitive
+It is a compile-time error if the class $C$
+does not have primitive equality
```

Comment:
This is where we actually only need primitive `operator ==`, because a switch is not expected to use `hashCode`.

I can see that it's simpler to just have one notion, but I think it can be relevant to have, say, an `enum` where `hashCode` defers to `index` for speed, rather than rely on `identitiyHashCode` (which is not that efficient), but where you still want to allow switching on the enum values.

Reply:
That is possible, but somewhat speculative: If two enum declarations do this then those objects could never be allowed to occur in the same set. I suspect that this would be a footgun that nobody would want to have in a serious software product.

Reply:
There should be no problem having them in the same set, as long as it's not a constant set literal.
An implementation of `int get hashCode => 4;` is *valid*, just not necessarily that efficient.